From 5dc634b67f3b1890c5f166694ba9b81dd0a5a871 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Fri, 19 May 2017 12:41:38 +0300
Subject: [PATCH] apalis-imx6: enable Fusion10 display if found during boot

Test i2c bus if Fusion10 display is present during boot.
If found, enable it as the primary display in kernel args.

Also remove console from tty1, as it overlays with Qt.
---
 include/configs/apalis_imx6.h | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/include/configs/apalis_imx6.h b/include/configs/apalis_imx6.h
index 4f2f689..de18b43 100644
--- a/include/configs/apalis_imx6.h
+++ b/include/configs/apalis_imx6.h
@@ -279,9 +279,14 @@
 		"|| setenv drive 2; mmc rescan; load ${interface} ${drive}:1" \
 		" ${loadaddr} flash_blk.img && " \
 		"source ${loadaddr}\0" \
-	"setup=setenv setupargs fec_mac=${ethaddr} " \
-		"consoleblank=0 no_console_suspend=1 console=tty1 " \
+	"setup=run setupvideo; setenv setupargs fec_mac=${ethaddr} " \
+		"consoleblank=0 no_console_suspend=1 " \
 		"console=${console},${baudrate}n8\0 " \
+	"setupvideo=i2c dev 0; if i2c probe 50; " \
+        "then setenv vidargs mxc_hdmi.only_cea=1 " \
+		"video=mxcfb0:dev=lcd,FusionF10A,if=RGB24 " \
+		"video=mxcfb1:off video=mxcfb2:off video=mxcfb3:off " \
+		"fbmem=32M; fi\0" \
 	"setupdate=run setsdupdate || run setusbupdate || run setethupdate\0" \
 	"setusbupdate=usb start && setenv interface usb; setenv drive 0; " \
 		"load ${interface} ${drive}:1 ${loadaddr} flash_blk.img && " \
